corehq/apps/linked_domain/updates.py

```diff
@@ -841,13 +850,13 @@ def _convert_reports_permissions(domain_link, master_results):
         role_def['permissions']['view_report_list'] = new_report_perms
 
 
-def _get_new_tableau_report_permissions(downstream_domain_visualizations, upstream_permissions,
-                                        original_downstream_permissions):
-    new_downstream_view_tableau_list = []
-    for viz in downstream_domain_visualizations:
-        upstream_viz_enabled = viz.upstream_id and viz.upstream_id in upstream_permissions.view_tableau_list
-        no_upstream_viz_and_viz_enabled_downstream = (not viz.upstream_id
-            and str(viz.id) in original_downstream_permissions.view_tableau_list)
-        if upstream_viz_enabled or no_upstream_viz_and_viz_enabled_downstream:
-            new_downstream_view_tableau_list.append(str(viz.id))
-    return new_downstream_view_tableau_list
+def _get_new_list_permissions(downstream_obj_list, upstream_permissions_list,
+                              current_downstream_permissions_list):
+    new_downstream_permissions_list = []
+    for obj in downstream_obj_list:
+        if obj.upstream_id:
```

Comment:
Do I understand the test correct, that it only tests this side of the if statement?